xpdo/xpdo.class.php

```diff
@@ -2034,12 +2034,31 @@ protected function _log($level, $msg, $target= '', $def= '', $file= '', $line= '
             if (!empty ($line)) {
                 $line= " : {$line}";
             }
+            switch (gettype($msg)) {
+                case 'array' :
+                    $log = print_r($msg, true);
+                    break;
+                case 'object' :
+                    if ($msg instanceof xPDOObject) {
+                        $log = print_r($msg->toArray(), true);
+                    } elseif ($msg instanceof xPDOQuery) {
```

Comment:
That's a good point. Would the only time an xPDOQuery get passed to the log function be manually? Just thinking about performance overall here.